Various improvements to fluentd-monasca plugin

markgoddard opened the pull request:
* Supports monasca log API V2.0 and V3.0
* Supports bulk submission of log messages to V3.0 API.

markgoddard committed:
First attempt at getting the monasca plugin working

Temporarily convert the plugin to an unbuffered output plugin as these
are simpler.

Supports monasca log API v2.0 and v3.0. All event record items other than
'message' are interpreted as dimensions.

markgoddard committed:
Convert out_monasca plugin to a buffered output

Log events are chunked and output asynchronously from the fluent engine.

markgoddard committed:
Support bulk writes using V3.0 log API

Monasca log API v3.0 supports POSTing multiple log messages in a single
request. When using the v3 API, use bulk message transmission.

markgoddard committed:
Update README with info on configuration

taimir commented:
All right, seems like you are interested in reviving this project. I'll do a quick review when I have some free time. In case you are wondering, it was left unfinished because [the logstash plugin](https://github.com/logstash-plugins/logstash-output-monasca_log_api) was given priority back at the time.

markgoddard commented:
Absolutely! We're currently using [kolla-ansible](https://github.com/openstack/kolla-ansible), which uses fluentd for log collection in the control plane. We use monasca for monitoring and log aggregation, and this plugin gave us a head start on integrating the two systems. Thanks.

markgoddard committed:
Fix use of logger during token validation

markgoddard committed:
Fix reauthentication on token expiration

markgoddard committed:
Add constraint of <0.14 for fluentd dependency

The gem fails to install in a kolla fluentd container at the pike
release without this dependency, with the following error:

ERROR:  Error installing fluentd-monasca-output-0.0.1.gem:
        serverengine requires Ruby version >= 2.1.0.
